pydantic/_internal/_metadata.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+import dataclasses
+from typing import Any, Iterable
+
+import annotated_types as at
+
+from ._fields import PydanticGeneralMetadata, PydanticMetadata
+
+SEQUENCE_CONSTRAINTS = {'min_length', 'max_length'}
+COMPARISON_CONSTRAINTS = {'le', 'ge', 'lt', 'gt'}
+NUMERIC_CONSTRAINTS = {'multiple_of', 'allow_inf_nan', *COMPARISON_CONSTRAINTS}
+
+CORE_SCHEMA_CONSTRAINTS = {
```

Comment:
I'm a bit lost on how some of this is used, e.g. I don't see anywhere that `CORE_SCHEMA_CONSTRAINTS['list']` is used.

Also, I'm not sure this should be a dictionary, surely it would be fine if these were each separate valiables?

Reply:
list is not being used but I added it just for completeness. happy to keep it to only the ones being used in this PR. also sure yeah separate variables works, it's just more imports if there ends up being a single module that uses them. And it's not like they're different types.

Reply:
let's have them all, but as variables.

Reply:
done